src/types/bytearray.rs

```diff
@@ -188,7 +188,9 @@ impl PyByteArray {
     /// }
     /// ```
     pub unsafe fn as_bytes(&self) -> &[u8] {
-        slice::from_raw_parts(self.data(), self.len())
+        let slice = Py2::borrowed_from_gil_ref(&self).as_bytes();
+        // SAFETY: &self guarantees the reference is alive long enough
+        unsafe { slice::from_raw_parts(slice.as_ptr(), slice.len()) }
```

Comment:
Couldn't these two use some form of `Py2Borrowed` as well to extend the lifetime of the slice?

Reply:
Yes, will fixup with clippy. 👍